firestore/angular-rewrite/src/app/homepage/homepage.component.ts

```diff
@@ -0,0 +1,33 @@
+import { Component } from '@angular/core';
+import { AngularFirestore } from '@angular/fire/compat/firestore';
+import { Router } from '@angular/router';
+import { Restaurant } from '../restaurant-card/restaurant';
+import { Observable } from 'rxjs';
+import { MatDialog } from '@angular/material/dialog';
+import { FilterDialogComponent } from '../filter-dialog/filter-dialog.component';
+import { DialogData } from '../filter-dialog/dialogdata';
+
+@Component({
+  selector: 'app-homepage',
+  templateUrl: './homepage.component.html',
+  styleUrls: ['./homepage.component.css']
+})
+
+export class HomepageComponent {
+  title = 'FriendlyEats-Homepage';
+  restaurants = this.store.collection('restaurants').valueChanges({ idField: 'id' }) as Observable<Restaurant[]>;
+
+  openDialog(): void {
+    const dialogRef = this.dialog.open(FilterDialogComponent, {
+      data: {
```

Comment:
nit: factor out to a constant and provide a name like "DEFAULT_SORT_DATA" or similar